includes/helpers/class-listing-image.php

```diff
@@ -138,4 +139,26 @@ public static function maybe_delete_image( $id, $listing_id ) {
 			delete_post_thumbnail( $listing_id );
 		}
 	}
+
+	/**
+	 * Checks id an attachment is linked to other posts.
+	 * This checks if the attachment is in use with other posts and returns the ids.
+	 *
+	 * @param int $attachment_id The attachment id.
+	 *
+	 * @since x.x
+	 *
+	 * @return array
+	 */
+	public static function get_linked_posts( $attachment_id ) {
```

Comment:
Let's make new functions private whenever possible to make future maintenance easier.